### PR TITLE
first try at issue 11

### PR DESCRIPTION
**Pull request by kassimbashir**
Relates #11

### CODE REVIEW

File: src/middleware/verifyToken.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+const secret = config.jwtSecret
+// TOKEN FORMAT: authorization: Bearer <access_token>
+const verifyToken = (
+    req: Request,
+    res: Response,
+    next: NextFunction
+)=>{
+    /** extracting token from header */ 
+        // get auth header value
+        const header = req.headers['authorization'];
```
Comment (Karyum) [inline src/middleware/verifyToken.ts:16]:
```suggestion
        const header = req.headers.authorization;
```

File: src/middleware/verifyToken.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        // check if header is  undefined
+        if(typeof header !== 'undefined'){
```
Comment (Karyum) [inline src/middleware/verifyToken.ts:18]:
```suggestion
        if(header){
```

File: src/middleware/verifyToken.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            // split at the space and get token from array
+            const token = header.split(' ')[1];
```
Comment (Karyum) [inline src/middleware/verifyToken.ts:20]: why is there a space ? does the header look like this ? `Bearer TOKEN22#*@(#@SDSD`

File: src/middleware/verifyToken.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+           // verify token
+           jwt.verify(token,secret,async (err,data)=>{
```
Comment (Karyum) [inline src/middleware/verifyToken.ts:22]: Change this to sync, no reason to be async and give it a callback

Example from docs:

```js
  var decoded = jwt.verify(token, 'SECRET');

```

File: src/middleware/verifyToken.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+               if(err){
+                   console.log(err);
+                   res.sendStatus(403);
+               }else{
+                   if(data){
+                    const user_id = data.id;
```
Comment (Karyum) [inline src/middleware/verifyToken.ts:28]:
```suggestion
                    const userId = data.id;
```

File: src/middleware/verifyToken.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+                     /** get user from DB using the id we got from verifying token */
+                    const result = await db.query('SELECT * FROM users WHERE id = $1', [user_id])
```
Comment (Karyum) [inline src/middleware/verifyToken.ts:30]:
```suggestion
                    const result = await db.query('SELECT * FROM users WHERE id = $1', [userId])
```

File: src/middleware/verifyToken.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+                    req.user = result.rows;
```
Comment (Karyum) [inline src/middleware/verifyToken.ts:31]:
```suggestion
                    req.user = result.rows[0];
```

**Review by Karyum — CHANGES_REQUESTED** [review]

File: src/middleware/verifyToken.ts (hunk 3 above)
Comment (kassimbashir) [reply src/middleware/verifyToken.ts:20]: yes it does

File: src/middleware/verifyToken.ts (hunk 4 above)
Comment (kassimbashir) [reply src/middleware/verifyToken.ts:22]: ok boss

**Review by Karyum — APPROVED** [review]

**Pull request merged**